src/lib/Components/Bidding/Screens/ConfirmBid.tsx

```diff
@@ -62,16 +68,18 @@ interface ConfirmBidState {
 const MAX_POLL_ATTEMPTS = 20
 
 const queryForBidPosition = (bidderPositionID: string) => {
-  return metaphysics({
+  // TODO: Auto-generate types specifically for this query rather than using an
```

Comment:
@yuki24 and I just synced about this IRL. We can do something similar to what we do in Prediction so that we can get relay's nice type generation without needing to actually couple this query to a GraphQL component.

Relevant prediction source that we looked at: https://github.com/artsy/prediction/blob/master/src/server/externals/metaphysics/GetSaleMeta.ts

Reply:
I've extracted the `BidderPositionQuery` out of the `<ConfirmBid>` component to generate the types for it: https://github.com/artsy/emission/pull/1744/files#diff-17aaff619e4681074c7b728228f47ae4

Reply:
I think we could take this one step further even. Relay exposes a `fetchQuery` function which means we don’t need the fake `GraphQL` function, we can retire the `metaphysics` function (I think); and potentially get some further benefits like cache invalidation in a single place, access to middlewares, etc.

But that’s probably all more for a follow-up PR.

Reply:
Very nice, it was a snap to switch to `fetchQuery`: https://github.com/artsy/emission/pull/1744/commits/097e3ba836a3c2a58a0122a45b6d5e1f7f5fd471